app/WebParsing/UtsgJsonParser.hs

```diff
@@ -0,0 +1,80 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+{-
+module WebParsing.UtsgJsonParser
+     (getJSON) where
+-}
+
+import Data.Aeson
+import Data.List
+import Database.Tables
+import WebParsing.PrerequisiteParsing
+import Database.Persist.Sqlite (runSqlite, insertMany_)
+import qualified Data.HashMap.Lazy as M
+import qualified Data.Text as T
+import Config (databasePath)
+import Network.HTTP.Conduit (simpleHttp)
+
+{- URL to UofT courses (stored as JSON string) -}
+jsonURL :: String
+jsonURL = "https://timetable.iit.artsci.utoronto.ca/api/courses?org=&code=csc108"
+
+{- Decode JSON string to JSON object -}
+getJSON :: IO (Maybe Object)
+getJSON = do
+    resp <- simpleHttp jsonURL
+    return $ (decode resp :: Maybe Object)
+
+{- Extract course metadata (values) from JSON object -}
+getMetadata :: IO (Maybe (M.HashMap k v)) -> IO [v]
+getMetadata = fmap (\(Just v) -> M.elems v)
```

Comment:
Couple things: 

1. You can use `$` after `fmap`.
2. Generally, if a function's type is `IO a -> IO b` and its top level is `fmap`, then I'd rather see the function defined as `a -> b` and then applied using `fmap`.
3. The pattern matching isn't great. If you really want to return a list, probably use the `maybe` function to either get `v` or an empty hash map.
But